File: 3264-final-array-state-after-k-multiplication-operations-i.cpp
/*
3264. Final Array State After K Multiplication Operations I

LeetCode Daily Question for December 15, 2024

Runtime: 0 ms (beats 100.00%)
Memory: 27.17 MB (beats 46.57%)
*/

class Solution {
public:
    vector<int> getFinalState(vector<int>& nums, int k, int multiplier) {
        for (int i = 0; i < k; ++i) {
            auto pos = nums.begin();
            for (auto it = nums.begin() + 1; it != nums.end(); ++it) {
                if (*it < *pos) pos = it;
            }
            *pos = *pos * multiplier;
        }
        return nums;
    }
};

File: 3264-final-array-state-after-k-multiplication-operations-i.py
"""
3264. Final Array State After K Multiplication Operations I

LeetCode Daily Question for December 15, 2024

Runtime: 255 ms (beats 5.80%)
Memory: 17.51 MB (beats 12.46%)
"""

class Solution:
    def getFinalState(self, nums: List[int], k: int, multiplier: int) -> List[int]:
        return reduce(
            lambda x, _: [x[i] * multiplier if i == (min(enumerate(x), key = (lambda e: e[1]))[0]) else x[i] for i in range(len(x))],
            range(k),
            nums
        )
